src/commands/filter_parser.cc

```diff
@@ -832,33 +870,58 @@ absl::StatusOr<std::unique_ptr<query::Predicate>> FilterParser::ParseExpression(
     }
     std::unique_ptr<query::Predicate> predicate;
     bool negate = Match('-');
-
     if (Match('(')) {
-      VMSDK_ASSIGN_OR_RETURN(predicate, ParseExpression(level));
+      VMSDK_ASSIGN_OR_RETURN(auto sub_result, ParseExpression(level));
       if (!Match(')')) {
         return absl::InvalidArgumentError(
             absl::StrCat("Expected ')' after expression got '",
                          expression_.substr(pos_, 1), "'. Position: ", pos_));
       }
-      if (prev_predicate) {
-        node_count_++;  // Count the ComposedPredicate Node
+      predicate = std::move(sub_result.prev_predicate);
+      // When there is no term inside the brackets
+      if (!predicate) {
+        return absl::InvalidArgumentError(
+            absl::StrCat("Empty brackets detected at Position: ", pos_ - 1));
+      }
+      if (result.prev_predicate) {
+        node_count_++;
       }
+      // If there is no Previous Predicate that means there is no term before
+      // it and hence it is the first group which should branch to a separate
+      // sub tree. This will be used when we encounter the next predicate with
+      // AND logical operator.
+      no_prev_grp = (!result.prev_predicate) ? true : false;
       VMSDK_ASSIGN_OR_RETURN(
-          prev_predicate,
-          WrapPredicate(std::move(prev_predicate), std::move(predicate), negate,
-                        query::LogicalOperator::kAnd));
+          result.prev_predicate,
+          WrapPredicate(std::move(result.prev_predicate), std::move(predicate),
+                        negate, query::LogicalOperator::kAnd, false,
+                        result.not_rightmost_bracket));
```

Comment:
`not_rightmost_bracket` is not used or needed when the operator is `query::LogicalOperator::kAnd`